src/utils/encryption.ts

```diff
@@ -1,83 +1,145 @@
     const cipherStream = new Readable({
-      read(): void { }
+      read(): void {},
     });
 
+    let buffer = Buffer.alloc(0);
+
     plaintextStream.on('data', (chunk) => {
-      const encryptedChunk = cipher.update(chunk);
-      cipherStream.push(encryptedChunk);
+      buffer = Buffer.concat([buffer, chunk]);
     });
 
-    plaintextStream.on('end', () => {
-      const finalChunk = cipher.final();
-      cipherStream.push(finalChunk);
-      cipherStream.push(null);
+    plaintextStream.on('end', async () => {
+      try {
+        // Encrypt the entire buffer when the stream ends
+        const encryptedData = await AesCtr.encrypt({
+          data    : buffer,
+          key     : jwkKey,
+          counter : initializationVector,
+          length  : 128, // FIX: Counter length must be between 1 and 128
+        });
+
+        cipherStream.push(encryptedData);
+        cipherStream.push(null); // Signal the end of the stream
+      } catch (error) {
+        cipherStream.emit('error', error); // Emit error if encryption fails
+      }
     });
 
     plaintextStream.on('error', (err) => {
-      cipherStream.emit('error', err);
+      cipherStream.emit('error', err); // Propagate errors
     });
 
-    return cipherStream;
+    return cipherStream; // Return the cipher stream
   }
 
-  /**
-   * Decrypts the given cipher stream using AES-256-CTR algorithm.
-   */
-  public static async aes256CtrDecrypt(key: Uint8Array, initializationVector: Uint8Array, cipherStream: Readable): Promise<Readable> {
-    const decipher = crypto.createDecipheriv('aes-256-ctr', key, initializationVector);
+  public static async aes256CtrDecrypt(
+    key: Uint8Array,
+    initializationVector: Uint8Array,
+    cipherStream: Readable
+  ): Promise<Readable> {
+    const jwkKey = await this.convertToJwk(key); // Convert key to JWK format
 
+    // Create a plaintext stream
     const plaintextStream = new Readable({
-      read(): void { }
+      read(): void {},
     });
 
+    let buffer = Buffer.alloc(0);
+
     cipherStream.on('data', (chunk) => {
-      const decryptedChunk = decipher.update(chunk);
-      plaintextStream.push(decryptedChunk);
+      buffer = Buffer.concat([buffer, chunk]);
     });
 
-    cipherStream.on('end', () => {
-      const finalChunk = decipher.final();
-      plaintextStream.push(finalChunk);
-      plaintextStream.push(null);
+    cipherStream.on('end', async () => {
+      try {
+        // Decrypt the entire buffer when the stream ends
+        const decryptedData = await AesCtr.decrypt({
+          data    : buffer,
+          key     : jwkKey,
+          counter : initializationVector,
+          length  : 128, // FIX: Counter length must be between 1 and 128
+        });
+
+        plaintextStream.push(decryptedData);
+        plaintextStream.push(null); // Signal the end of the stream
+      } catch (error) {
+        plaintextStream.emit('error', error); // Emit error if decryption fails
```

Comment:
Sanity: This line isn't covered by test coverage, let's write a test that validates the behavior.